src/astarte/handler.rs

```diff
@@ -321,6 +320,44 @@ impl DevicePublisher {
 
         Ok(())
     }
+
+    // check if the interface is present in the node introspection
+    async fn find_in_introspection(
```

Comment:
nit: otherwise, it would return the interface or a boolean

```suggestion
    async fn check_node_has_interface(
```